components/src/AppModalLogin/appmodallogin.main.tsx

```diff
@@ -87,8 +87,14 @@ class AppModalLoginMain extends Component<AppModalLoginMainProps, AppModalLoginM
     const url = locationSearchData;
     const params = queryString.parse(url);
     if (params.code) {
-      localStorage.setItem(`${Config.cortexApi.scope}_keycloakCode`, params.code);
-      localStorage.setItem(`${Config.cortexApi.scope}_keycloakSessionState`, params.session_state);
+      if (Config.b2b.openId.enabled) {
+        localStorage.setItem(`${Config.cortexApi.scope}_openIdcCode`, params.code);
+        localStorage.setItem(`${Config.cortexApi.scope}_openIdcSessionState`, params.session_state);
+      } else {
+        localStorage.setItem(`${Config.cortexApi.scope}_keycloakCode`, params.code);
+        localStorage.setItem(`${Config.cortexApi.scope}_keycloakSessionState`, params.session_state);
+      }
+      localStorage.removeItem('OidcSecret');
       if (localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) !== 'REGISTERED') {
         loginRegisteredAuthService(params.code, encodeURIComponent(Config.b2b.keycloak.callbackUrl), encodeURIComponent(Config.b2b.keycloak.client_id)).then((resStatus) => {
```

Comment:
in this line we'll want to check if open-id feature is enabled, and make sure to set the parameters for the callback and client_id for that if so